src/tests/ftest/tags.py

```diff
@@ -650,90 +747,211 @@ def print_verbose(*args):
+    dump_parser.add_argument(
+        "--paths",
+        nargs="+",
+        default=[],
+        help="file paths")
+    dump_parser.add_argument(
+        "--tags",
+        nargs="+",
+        type=__comma_separated_arg,
+        help="tags used to filter which files to display")
+
+    # Parser for the "lint" command and its optional arguments
+    lint_parser = subparsers.add_parser(
+        'lint',
+        help='lint ftest avocado tags')
+    lint_parser.add_argument(
+        "--paths",
+        nargs="+",
+        default=[],
+        help="file paths")
+    lint_parser.add_argument(
         "-v", "--verbose",
         action='store_true',
         help="print verbose output for some commands")
-    parser.add_argument(
+
+    # Parser for the "list" command and its optional arguments
+    list_parser = subparsers.add_parser(
+        'list',
+        help='list ftest avocado tags associated with test files')
+    list_parser.add_argument(
         "--paths",
         nargs="+",
         default=[],
         help="file paths")
-    parser.add_argument(
-        "--tags",
-        nargs="+",
-        type=__arg_type_tags,
-        help="tags")
+
+    # Parser for the "unit" command and its optional arguments
+    unit_parser = subparsers.add_parser(
+        'unit',
+        help='run self unit tests')
+    unit_parser.add_argument(
+        "-v", "--verbose",
+        action='store_true',
+        help="print verbose output")
+
     args = parser.parse_args()
-    args.paths = list(map(os.path.realpath, args.paths))
-
-    # Check for incompatible arguments
-    rc = 0
-    if args.command == "lint" and args.tags:
-        print("--tags not supported with lint")
-        rc = 1
-    if args.command == "list" and args.tags:
-        print("--tags not supported with list")
-        rc = 1
-    if args.command == "unit" and args.tags:
-        print("--tags not supported with unit")
-        rc = 1
-    if args.command == "unit" and args.paths:
-        print("--paths not supported with unit")
-        rc = 1
-    if rc != 0:
-        return rc
-
-    if args.command == "lint":
+    if hasattr(args, "paths"):
+        args.paths = list(map(os.path.realpath, args.paths))
+
+    if args.command == "config":
+        rc = run_config(args.paths, args.tags, args.keys, args.csv)
+    elif args.command == "dump":
+        rc = run_dump(args.paths, args.tags)
+    elif args.command == "lint":
         try:
             run_linter(args.paths, args.verbose)
             rc = 0
         except LintFailure as err:
             print(err)
             rc = 1
-    elif args.command == "dump":
-        rc = run_dump(args.paths, args.tags)
     elif args.command == "list":
         rc = run_list(args.paths)
     elif args.command == "unit":
         test_tag_set()
         test_tags_util(args.verbose)
         rc = 0
+    else:
+        raise ValueError(f"Unhandled command option: {args.command}")
```

Comment:
I get this
```
./tags.py 
Traceback (most recent call last):
  File "/home/dbohning/daos-stack/daos/src/tests/ftest/./tags.py", line 960, in <module>
    sys.exit(main())
  File "/home/dbohning/daos-stack/daos/src/tests/ftest/./tags.py", line 954, in main
    raise ValueError(f"Unhandled command option: {args.command}")
ValueError: Unhandled command option: None
```

Reply:
Fixed by making subparser 'command' required.